dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -396,6 +509,15 @@ void WindowTransformAction::advanceFrameEnd()
         break;
     }
     case WindowFrame::BoundaryType::Offset:
+    {
+        constexpr size_t frame_end_pos = 0;
+        constexpr size_t is_frame_end_pos = 1;
+        auto res = stepBackward(current_row, window_description.frame.end_offset);
+        frame_ended = std::get<is_frame_end_pos>(res);
+        if (frame_ended)
+            frame_end = std::get<frame_end_pos>(res);
```

Comment:
It seems that when A is false, the value of B is meaningless, so just assign it directly?
```suggestion
        std::tie(frame_end, frame_ended) = stepBackward(current_row, window_description.frame.end_offset);
```

Reply:
> It seems that when A is false, the value of B is meaningless, so just assign it directly?

okk